azurerm/resource_arm_application_gateway.go

```diff
@@ -246,6 +246,24 @@ func resourceArmApplicationGateway() *schema.Resource {
 							},
 						},
 
+						"trusted_root_certificate": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+
+									"id": {
```

Comment:
Do you see anyone using this `id` anywhere? If not, it might be worth removing this block entirely and just having an attribute called `trusted_root_certificate_name`

Reply:
Hi @mbfrahry, you're right, I don't think id is needed, a list of strings should be enough. I pushed a new commit with your suggestion and managed the state. Let me know what you think. Unit test is still missing, I'll try to provide one asap.